src/main/java/uk/gov/hmcts/reform/sscs/helper/service/HearingsServiceHelper.java

```diff
@@ -1,34 +1,82 @@
 package uk.gov.hmcts.reform.sscs.helper.service;
 
-import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import org.jetbrains.annotations.Nullable;
+import uk.gov.hmcts.reform.sscs.ccd.domain.Hearing;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.HearingState;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
 import uk.gov.hmcts.reform.sscs.model.HearingEvent;
 import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
 import uk.gov.hmcts.reform.sscs.model.single.hearing.HmcUpdateResponse;
 
-import java.util.Optional;
+import java.util.ArrayList;
+import javax.validation.Valid;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
 
 public final class HearingsServiceHelper {
 
     private HearingsServiceHelper() {
     }
 
-    public static void updateHearingId(HearingWrapper wrapper, HmcUpdateResponse response) {
-        wrapper.getCaseData().getSchedulingAndListingFields().setActiveHearingId(response.getHearingRequestId());
+    public static void updateHearingId(Hearing hearing, HmcUpdateResponse response) {
+        if (nonNull(response.getHearingRequestId())) {
+            hearing.getValue().setHearingId(String.valueOf(response.getHearingRequestId()));
+        }
     }
 
-    public static void updateVersionNumber(HearingWrapper wrapper, HmcUpdateResponse response) {
-        wrapper.getCaseData().getSchedulingAndListingFields().setActiveHearingVersionNumber(response.getVersionNumber());
+    public static void updateVersionNumber(Hearing hearing, HmcUpdateResponse response) {
+        hearing.getValue().setHmcVersionNumber(response.getVersionNumber());
     }
 
     public static HearingEvent getHearingEvent(HearingState state) {
         return HearingEvent.valueOf(state.name());
     }
 
     public static String getHearingId(HearingWrapper wrapper) {
-        return Optional.of(wrapper)
-            .map(HearingWrapper::getCaseData)
-            .map(SscsCaseData::getSchedulingAndListingFields)
-            .map(SchedulingAndListingFields::getActiveHearingId)
-            .map(Object::toString).orElse(null);
+        Hearing hearing = wrapper.getCaseData().getLatestHearing();
+        if (nonNull(hearing)) {
+            return hearing.getValue().getHearingId();
+        }
+        return null;
+    }
+
+    public static Long getVersion(HearingWrapper wrapper) {
+        Hearing hearing = wrapper.getCaseData().getLatestHearing();
+        if (nonNull(hearing)) {
+            Long version = hearing.getValue().getHmcVersionNumber();
+            if (nonNull(version) && version > 0) {
+                return version;
+            }
+        }
+        return null;
+    }
+
+    public static Hearing findOrCreateHearingInCaseData(Long hearingId, @Valid SscsCaseData caseData) {
```

Comment:
I'm unsure about keeping the create hearing in this method, it's used in multiple places - not all of which should create a hearing if one isn't found, should they?